cmd/data-node/unsafe_reset_all.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2022 Gobalsky Labs Limited
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at https://www.mariadb.com/bsl11.
+//
+// Change Date: 18 months from the later of the date of the first publicly
+// available Distribution of this version of the repository, and 25 June 2022.
+//
+// On the date above, in accordance with the Business Source License, use
+// of this software will be governed by version 3 or later of the GNU General
+// Public License.
+
+package main
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	"code.vegaprotocol.io/shared/paths"
+	"code.vegaprotocol.io/vega/config"
+	"code.vegaprotocol.io/vega/logging"
+
+	"github.com/jessevdk/go-flags"
+)
+
+type UnsafeResetAllCmd struct {
+	config.VegaHomeFlag
+}
+
+func (cmd *UnsafeResetAllCmd) Execute(_ []string) error {
+	log := logging.NewLoggerFromConfig(
+		logging.NewDefaultConfig(),
+	)
+	defer log.AtExit()
+
+	vegaPaths := paths.New(cmd.VegaHome)
+
+	storagePath := vegaPaths.StatePathFor(paths.DataNodeStorageHome)
+	if err := deleteAll(log, storagePath); err != nil {
+		log.Error("Unable to remove storage state", logging.Error(err))
+	} else {
+		log.Info("Removed storage state", logging.String("path", storagePath))
+	}
+	return nil
+}
+
+func deleteAll(log *logging.Logger, dir string) error {
+	d, err := os.Open(dir)
+	if err != nil {
+		return err
+	}
```

Comment:
```suggestion
	}
	defer d.Close()
```
But here program anyway is closed shortly so not required.

Reply:
it is definitely required, that was a mistaken of mine